src/pal/inc/pal.h

```diff
@@ -195,6 +195,8 @@ typedef __builtin_va_list va_list;
 
 #define PUB __attribute__((visibility("default")))
 
+#define __sync_swap(handler, func) __atomic_exchange_n(handler, func, __ATOMIC_ACQ_REL)
```

Comment:
I would prefer replacing the two usages of `__sync_swap` in pal.h by the `__atomic_exchange_n` instead.